### PR TITLE
ENH: DataFrame.loc now supports 2d indexing

### PR DESCRIPTION
**Pull request by sinhrks**
`DataFrame.loc` now supports 2d indexing like `df.loc[5:10, 'a']`.

### CODE REVIEW

File: dask/dataframe/indexing.py [hunk 1]
```diff
@@ -2,10 +2,162 @@
...
+            return self.obj
+        else:
+            return self.obj._meta.loc[:, cindexer]
+
+    def __getitem__(self, key):
+
+        if isinstance(key, tuple):
+            # multi-dimensional selection
+            if len(key) > self.obj.ndim:
+                raise KeyError(key)
```
Comment (TomAugspurger) [inline dask/dataframe/indexing.py]: Is this like

```python
In [13]: df = pd.DataFrame(np.random.randn(10, 2))

In [14]: df.loc[(0, 1, 'c')]
```

Would we want to raise an `IndexingError` here to be consistent with pandas?

**Comment by sinhrks** [comment]
@TomAugspurger updated to raise ``IndexingError``.  Though ``pandas`` may raise ``KeyError`` sometimes, the PR always uses ``IndexingError`` for consistency.

```
pd.Series([1, 2, 3]).loc[3, 4]
# KeyError: 'the label [3] is not in the [index]'

pd.Series([1, 2, 3]).loc[1, 2, 3]
# pandas.core.indexing.IndexingError: Too many indexers
```

**Comment by mrocklin** [comment]
Any further comments here?

**Comment by TomAugspurger** [comment]
+1 from me

**Comment by mrocklin** [comment]
Great.  Merging.  @TomAugspurger @sinhrks Please feel free to merge in the future.

**Pull request merged**